src/databricks/labs/dqx/col_functions.py

```diff
@@ -452,3 +424,42 @@ def is_valid_timestamp(col_name: str, timestamp_format: str | None = None) -> Co
         F.concat_ws("", F.lit("Value '"), column, F.lit(condition_str)),
         f"{col_name}_is_not_valid_timestamp",
     )
+
+
+def is_unique(col_name: str) -> Column:
+    """Checks whether the values in the input column are unique
+    and reports an issue for each row that contains a duplicate value.
+    Null values are not considered duplicates, following the ANSI SQL standard.
```

Comment:
Should we put some warning about use in streaming?
Should we allow to make bring their own window definition? Like, unique value during the hour, or something like that?

Reply:
added comment regarding streaming and added window_spec as optional argument.